python/ray/train/_internal/data_config.py

```diff
@@ -111,14 +110,14 @@ def configure(
     def default_ingest_options() -> ExecutionOptions:
         """The default Ray Data options used for data ingest.
 
-        We enable output locality, which means that Ray Data will try to place tasks on
-        the node the data is consumed. We also set the object store memory limit
-        to a fixed smaller value, to avoid using too much memory per Train worker.
+        By default, output locality is enabled, which means that Ray Data will try to
+        place tasks on the node the data is consumed. The remaining configurations are
+        carried over from what is already set in DataContext.
         """
         ctx = ray.data.DataContext.get_current()
         return ExecutionOptions(
             locality_with_output=True,
-            resource_limits=ExecutionResources(object_store_memory=2e9),
+            resource_limits=ctx.execution_options.resource_limits,
```

Comment:
I think previously has a smaller limit, makes sense for dataset not split (validation dataset each has its own memory limit per worker).

But right now, we split train and validation dataset by default, so it's no longer an issue.

Reply:
Cheng, it is absolutely still an issue, because splitting is not mandatory. After this change, ingest will likely spill/OOM whenever splitting is off, or if there are multiple concurrent trials being run by Tune.

We have to be careful about edge cases for these defaults.